tests/utils.go

```diff
@@ -320,16 +331,24 @@ func (w *ObjectEventWatcher) Watch(ctx context.Context, processFunc ProcessFunc,
 
 	if w.failOnWarnings {
 		f = func(event *k8sv1.Event) bool {
+			isIgnored := false
+			for _, message := range w.warningsIgnoreList {
+				if message == event.Message {
+					isIgnored = true
+					break
+				}
+			}
+
 			msg := fmt.Sprintf("Event(%#v): type: '%v' reason: '%v' %v", event.InvolvedObject, event.Type, event.Reason, event.Message)
-			if event.Type == string(WarningEvent) {
+			if event.Type == string(WarningEvent) && isIgnored == false {
 				log.Log.Reason(fmt.Errorf("unexpected warning event received")).ObjectRef(&event.InvolvedObject).Error(msg)
+				ExpectWithOffset(1, event.Type).NotTo(Equal(string(WarningEvent)), "Unexpected Warning event received: %s,%s: %s", event.InvolvedObject.Name, event.InvolvedObject.UID, event.Message)
 			} else {
 				log.Log.ObjectRef(&event.InvolvedObject).Info(msg)
 			}
-			ExpectWithOffset(1, event.Type).NotTo(Equal(string(WarningEvent)), "Unexpected Warning event received: %s,%s: %s", event.InvolvedObject.Name, event.InvolvedObject.UID, event.Message)
+
```

Comment:
This can be simplified like this 
```golang
	 if shouldNotIgnoreThisWarning {
              ExpectWithOffset(...
         }
	log.Log.ObjectRef(&event.InvolvedObject).Info(msg)
```